Create T+C, privacy policy, code of conduct pages

weepotty opened the pull request:
this PR creates the T+C, privacy policy, code of conduct pages with placeholder content.

I've yet to add in the links from the home page, will do that when we have the content / if the URL path names are ok.

ayushn21 commented:
I reckon we can avoid data files for this. All 3 pages should look pretty similar so I think we can create a single layout called "legal" or something like that, and then use plain markdown for the copy in the pages. I'll leave this open for now though until we have the copy for the pages so we can make a more informed decision then.

weepotty commented:
Hey @ayushn21, updated the copy for T+C and CoC pages, and linked to privacy policy as Amanda requested.
Also made the changes to the callout banner for mobile.
I've also added `target=_blank` to the portfolio and social links in the footer.

ayushn21 commented on _layouts/world/legal.html:
```diff
@@ -0,0 +1,8 @@
+---
+
+<content-container class="legal">
+<h1>{{ page.title }}</h1>
```
Could you fix the indentation here please? 

The industry standard is to use spaces over tabs (opinions on this could start a war though hahaha), and to use 2 spaces instead of a "tab". Pretty much every project I've ever worked on has had this convention. You should be able to configure this in your text editor.

ayushn21 commented on _layouts/world/legal.html:
```diff
@@ -0,0 +1,8 @@
+
+<content-container class="legal">
+<h1>{{ page.title }}</h1>
+    {{ content | markdownify }}
```
We don't need to explicitly add `markdownify` here since the file has the `md` extension. It'll automatically processed through a markdown converter.

ayushn21 commented on _sass/world/modules/_legal.scss:
```diff
@@ -0,0 +1,22 @@
+  }
+
+  p {
+    font-variation-settings: 'wght' 400, 'wdth' 100, 'slnt' 0;
```
Can we use a CSS variable here? I created `--font-weight-400-normal` yesterday.

weepotty replied:
I set the slant to 0 in this one because it looked a bit much with an entire block of text in italics. Didn't make a variable cos only anticipate using it for this case

ayushn21 replied:
Yeah I agree about the styling, I did the same thing for the FAQ answers. A variable for this already exists, so let's use it :)

weepotty replied:
ah yes i missed that! will use it

ayushn21 commented on world/conduct.md:
```diff
@@ -0,0 +1,47 @@
+
+As a Rails World attendee, I pledge to:
+
+Treat all community members with respect, kindness, and empathy.
```
Let's put all these points in a list. You can do that in Markdown by adding a `*` at the start of each point. You'll need to set `list-style-type: disc` targeting `ul` in legal content as well.

ayushn21 commented on world/terms.md:
```diff
@@ -0,0 +1,105 @@
+
+## **Registration and Attendance:** ##
+
+a. To attend our events, please make sure to register in advance through our official registration process. We won't be selling tickets at the door, so don't forget to register early!
```
Can we use an `ol` here? And below as well. There's a markdown way to do it but can't remember the syntax off the top of my head.

weepotty replied:
i can't find a way in markdown to have "a., b., c.," as the ordering style, that doesn't involve using HTML ie `<ol type="a">`. Should i use that?

ayushn21 replied:
We can combine markdown with CSS here to get the a, b, c ordinals. We can render a basic ordered list by writing 1., 2., 3. and so on in the markdown:

https://www.markdownguide.org/basic-syntax#ordered-lists

And then to use letters in the rendered output, we can set `list-style-type` in CSS targeting that list:

https://www.w3schools.com/CSS/tryit.asp?filename=trycss_list-style-type_ex

ayushn21 reviewed (approved):
A few things to address but mostly good!

ayushn21 commented on _sass/world/modules/_callout.scss:
```diff
@@ -26,10 +26,11 @@
       border-top: var(--border-red);
       border-bottom: var(--border-red);
       align-items: center;
+      justify-content: center;
```
Could you remove this commit please? My bad, I thought we could use `flex-wrap` without a media query so it just wraps when it needs to, but that doesn't look good.

The advantage of `flex-wrap` is that you don't need to put it in a media query, it just wraps when it needs to, but for this we'll need to specify some additional styles to make it look good. I'm experimenting with it and will PR separately.

weepotty commented:
i've made those changes to the markdown files plus the styling ones from our standup.. if it's ok i'll merge :)

ayushn21 commented:
👏
